cmd/kubeapps-apis/plugins/resources/v1alpha1/namespaces_filtering.go

```diff
@@ -0,0 +1,184 @@
+// Copyright 2022 the Kubeapps contributors.
+// SPDX-License-Identifier: Apache-2.0
+package main
+
+import (
+	"context"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+	authorizationapi "k8s.io/api/authorization/v1"
+	corev1 "k8s.io/api/core/v1"
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	log "k8s.io/klog/v2"
+	"math"
+	"regexp"
+	"strings"
+	"sync"
+)
+
+type checkNSJob struct {
+	ns corev1.Namespace
+}
+
+type checkNSResult struct {
+	checkNSJob
+	allowed bool
+	Error   error
+}
+
+func (s *Server) MaxWorkers() int {
+	return int(s.clientQPS)
+}
+
+// GetAccessibleNamespaces return the list of namespaces that the user has permission to access
+func (s *Server) GetAccessibleNamespaces(ctx context.Context, cluster string, trustedNamespaces []corev1.Namespace) ([]corev1.Namespace, error) {
+	var namespaceList []corev1.Namespace
+
+	if len(trustedNamespaces) > 0 {
+		namespaceList = append(namespaceList, trustedNamespaces...)
+	} else {
+
+		typedClient, _, err := s.clientGetter(ctx, cluster)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "unable to get the k8s client: '%v'", err)
+		}
+
+		// Try to list namespaces with the user token, for backward compatibility
+		namespaces, err := typedClient.CoreV1().Namespaces().List(context.TODO(), metav1.ListOptions{})
```

Comment:
I don't understand why `context.TODO()` is used here. Shouldn't we pass the parent function's context `ctx` or `context.Background()` in case we want a totally new context?

Reply:
The auth token of the request is actually handled in the `s.clientGetter` func of line 44. It is, in turn, handled [here](https://github.com/vmware-tanzu/kubeapps/blob/main/cmd/kubeapps-apis/core/plugins/v1alpha1/plugins.go#L319) based on the request's context.
Commenting here your post below: parts of the code are legacy from Kubeops so we inherit some stuff, but agree with you that we should try to harmonize the usage of `context.Background()` and `context.TODO()`. The two are mainly the same (see the [contexts definition here](https://go.dev/src/context/context.go#L199)), it is regarding semantics.

Go docs say that `Code should use context.TODO when it's unclear which Context to use or it is not yet available (because the surrounding function has not yet been extended to accept a Context parameter).`

[Documentation here](https://pkg.go.dev/context#Background).

Reply:
I see, thanks for the explanation 🙂